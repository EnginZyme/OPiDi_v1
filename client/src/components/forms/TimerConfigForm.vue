<template>
  <div class="step_config_form" v-if="formType === 'timer'">
    <p class="step_config_form__title">Timer</p>
    <img
      src="../../assets/coming_soon_meme.jpg"
      style="border: none; width: 100%;"
      alt="Coming soon!"
    />
  </div>
</template>

<script>
/**
 * Accepts user input on the configuration of a user selected timer step within the protocol cache object.
 */
export default {
  name: "TimerConfigForm",
  props: {
    /**
     * Determines if this component will be rendered or not (true if its value is set to `timer`).
     */
    formType: String,
    /**
     * Indicates the index of the user selected step card within the protocol editor UI.
     */
    stepID: Number,
  },
};
</script>

<style scoped>
.step_config_form {
  display: grid;
  grid-template-rows: auto auto;
  padding-top: 20px;
  padding-bottom: 10px;
}
.step_config_form__title {
  font-size: 18px;
  margin: 0px;
}
</style>
