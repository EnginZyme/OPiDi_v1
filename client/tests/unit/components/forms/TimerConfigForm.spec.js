import { shallowMount } from "@vue/test-utils";
import TimerConfigForm from "@/components/forms/TimerConfigForm"

const sampleFormType = "timer";
const sampleStepID = -1;

describe("shallowMounted TimerConfigForm", () => {
    const wrapper = shallowMount(TimerConfigForm, {
      propsData: {
        formType: sampleFormType,
        stepID: sampleStepID,
      },
    });

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})
