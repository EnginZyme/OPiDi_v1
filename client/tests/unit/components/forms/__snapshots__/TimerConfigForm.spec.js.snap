// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted TimerConfigForm renders correctly 1`] = `
<div
  class="step_config_form"
>
  <p
    class="step_config_form__title"
  >
    Timer
  </p>
   
  <img
    alt="Coming soon!"
    src="../../assets/coming_soon_meme.jpg"
    style="width: 100%;"
  />
</div>
`;
